### PR TITLE
RavenDB_19548: edited SelectMemberAccess and TranslateSelectBodyToJs methods to include Raw RavenQuery without 'new' keyword

### PR DESCRIPTION
**Pull request by TheGoldenPlatypus**
…methods to include Raw RavenQuery without 'new' keyword

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19548/When-using-RavenQuery.Raw-without-select-new-...-the-query-dosent-include-the-raw-Raw-content

### Additional description

edited SelectMemberAccess and TranslateSelectBodyToJs methods to include Raw RavenQuery without 'new' keyword

### Type of change

- Bug fix

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: test/SlowTests/Issues/RavenDB_19548.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new TestObj { Name = "Omer", Prop1 = "123456", Prop2 = "12", Birthday = new DateTime(1994, 3, 22) });
+                    await session.SaveChangesAsync();
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>() select result.Prop1.Length;
+                    Console.WriteLine(asyncDocumentQuery);
```
Comment (ayende) [inline test/SlowTests/Issues/RavenDB_19548.cs]: Remove this

File: test/SlowTests/Issues/RavenDB_19548.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                }
+
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>()
+                                             let ret = RavenQuery.Raw<int>("result.Prop1.length")
+                                             select ret;
+
+                    Console.WriteLine(asyncDocumentQuery);
```
Comment (ayende) [inline test/SlowTests/Issues/RavenDB_19548.cs]: Remove

File: test/SlowTests/Issues/RavenDB_19548.cs (hunk 1 above)
Comment (ayende) [reply test/SlowTests/Issues/RavenDB_19548.cs]: Add an assert on the query text, so it is easier to see what it is doing here.
(And in all the other locations as well)

Comment (TheGoldenPlatypus) [reply test/SlowTests/Issues/RavenDB_19548.cs]: Done

File: test/SlowTests/Issues/RavenDB_19548.cs (hunk 2 above)
Comment (TheGoldenPlatypus) [reply test/SlowTests/Issues/RavenDB_19548.cs]: Done

File: src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs [hunk 3]
```diff
@@ -3009,6 +3011,9 @@ private string TranslateSelectBodyToJs(Expression expression)
                     
                 }
             }
+            
+            if (expression is MemberExpression memberExpression)
+                AddJsProjection(memberExpression.Member.Name, memberExpression, sb, false);
```
Comment (garayx) [inline src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs]:
```suggestion
                AddJsProjection(memberExpression.Member.Name, memberExpression, sb, addComma: false);
```

File: test/SlowTests/Issues/RavenDB_19548.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+namespace SlowTests.Issues
+{
+    public class RavenDB_19548 : RavenTestBase
+    {
+        public RavenDB_19548(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task TestCase()
```
Comment (garayx) [inline test/SlowTests/Issues/RavenDB_19548.cs]: Please add meaningful name for test

File: test/SlowTests/Issues/RavenDB_19548.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+                    var queryResult = await asyncDocumentQuery.ToListAsync();
+                    Assert.Equal("Ome", queryResult[0]);
+                    Assert.Equal("declare function output(result) {\r\n\tvar ret = result.Name.substr(0,3);\r\n\treturn { ret : ret };\r\n}\r\nfrom 'TestObjs' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+                }
+
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<TestObj>()
```
Comment (garayx) [inline test/SlowTests/Issues/RavenDB_19548.cs]: do we want to add support for the following query?

```
var asyncDocumentQuery = from result in session.Query<TestObj>()
                select RavenQuery.Raw<int>("result.Prop1.length");
```

File: src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs (hunk 3 above)
Comment (TheGoldenPlatypus) [reply src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs]: Done

File: test/SlowTests/Issues/RavenDB_19548.cs (hunk 4 above)
Comment (TheGoldenPlatypus) [reply test/SlowTests/Issues/RavenDB_19548.cs]: Done

File: test/SlowTests/Issues/RavenDB_19548.cs (hunk 5 above)
Comment (TheGoldenPlatypus) [reply test/SlowTests/Issues/RavenDB_19548.cs]: Done

File: src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs [hunk 6]
```diff
@@ -2999,19 +3016,26 @@ private string TranslateSelectBodyToJs(Expression expression)
...
 
+            if (expression is MemberExpression or MethodCallExpression)
+            { 
+                if (IsRawOrTimeSeriesCall(expression, out string script) == false)
+                    script = ToJs(expression);
+                
+                sb.Append(script);
+                return sb.ToString();
+            }
             sb.Append(" }");
```
Comment (aviv86) [inline src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs]: move `sb.Append(" }")` inside the `if` clauses

File: test/SlowTests/Issues/RavenDB_19548.cs [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+                {
+                    var asyncDocumentQuery = from result in session.Query<User>()
+                        let ret = RavenQuery.Raw<object>("result.Id.length")
+                        select ret;
+
+                    Assert.Equal(
+                        "declare function output(result) {\r\n\tvar ret = result.Id.length;\r\n\treturn ret;\r\n}\r\nfrom 'Users' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+
+                }
```
Comment (aviv86) [inline test/SlowTests/Issues/RavenDB_19548.cs]: assert that query execution throws the expected error

File: test/SlowTests/Issues/RavenDB_19548.cs [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+                 */
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<User>()
+                        select RavenQuery.Raw<object>("result.Id.length");
+
+                    Assert.Equal(
+                        "declare function output(result) {\r\n\treturn result.Id.length;\r\n}\r\nfrom 'Users' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+                }
```
Comment (aviv86) [inline test/SlowTests/Issues/RavenDB_19548.cs]: same, assert that query throws

**Review by aviv86 — COMMENTED** [review]
LGTM

File: src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs (hunk 6 above)
Comment (TheGoldenPlatypus) [reply src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs]: Done

File: test/SlowTests/Issues/RavenDB_19548.cs (hunk 7 above)
Comment (TheGoldenPlatypus) [reply test/SlowTests/Issues/RavenDB_19548.cs]: Done

File: test/SlowTests/Issues/RavenDB_19548.cs (hunk 8 above)
Comment (TheGoldenPlatypus) [reply test/SlowTests/Issues/RavenDB_19548.cs]: Done

**Comment by TheGoldenPlatypus** [comment]
test this please

**Comment by TheGoldenPlatypus** [comment]
test this please

**Review by garayx — APPROVED** [review]

**Pull request merged**